### DISCUSSION TITLE
ID gets typed as string | number

### DISCUSSION THREAD

**Discussion by VinceBT-BG** [0no-co/gql.tada#117] [Q&A] [answered]
### Describe the bug

The scalar `ID` type gets typed as `string | number` when using `gql.tada`, meanwhile while using some basic library to generate typescript types like graphql-codegen, it types it as `string`.
The generated type by `gql.tada`:
<img width="425" alt="Capture d’écran 2024-03-07 à 18 13 49" src="https://github.com/0no-co/gql.tada/assets/117643868/30cf3643-cec5-4b0a-b5e7-5fc4cb858a1e">
The type generated by `graphql-codegen`
<img width="301" alt="Capture d’écran 2024-03-07 à 18 14 02" src="https://github.com/0no-co/gql.tada/assets/117643868/30fb3ee7-fa9a-4c7b-a86c-c07d4c1f69a0">
I am not sure which one is right, but this is something that I found on related to this matter: 
> The ID type is serialized in the same way as a String; however, it is not intended to be human‐readable. While it is often numeric, it should always serialize as a String

### gql.tada version

1.3.1

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/0no-co/gql.tada/discussions) should be used
- [X] Read the [docs](https://gql-tada.0no.co/).
- [X] Follow our [Code of Conduct](https://github.com/0no-co/gql.tada/blob/main/CODE_OF_CONDUCT.md)

**Comment by JoviDeCroock** [comment, marked as answer]
You can adjust this yourself by doing

```
initGraphQLTada<{ scalars: { ID: string }}>()
```

By default however, ID is an opaque type and it's correct to say `string | number`

**Reply by kitten** [reply]
This default has also now changed to just `string`
